climada/engine/unsequa/calc_delta_climate.py

```diff
@@ -0,0 +1,457 @@
+
+import pandas as pd
+import numpy as np
+import pathos.multiprocessing as mp
+# use pathos.multiprocess fork of multiprocessing for compatibility
+# wiht notebooks and other environments https://stackoverflow.com/a/65001152/12454103
+
+from climada.engine import ImpactCalc
+from climada.engine.unsequa import Calc, InputVar, UncImpactOutput
+from climada.engine.unsequa.calc_base import (
+    _sample_parallel_iterator,
+    _multiprocess_chunksize,
+    _transpose_chunked_data,
+)
+from climada.entity import Exposures, ImpactFuncSet
+from climada.hazard import Hazard
+from climada.util import log_level
+
+LOGGER = logging.getLogger(__name__)
+
+class CalcDeltaImpact(Calc):
+    """
+    Impact uncertainty caclulation class.
+
+    This is the class to perform uncertainty analysis on the outputs of a
+    climada.engine.impact.Impact() object.
+
+    Attributes
+    ----------
+    rp : list(int)
+        List of the chosen return periods.
+    calc_eai_exp : bool
+        Compute eai_exp or not
+    calc_at_event : bool
+        Compute eai_exp or not
+    value_unit : str
+        Unit of the exposures value
+    exp_input_var : InputVar or Exposures
+        Exposure uncertainty variable
+    impf_input_var : InputVar if ImpactFuncSet
+        Impact function set uncertainty variable
+    haz_input_var: InputVar or Hazard
+        Hazard uncertainty variable
+    _input_var_names : tuple(str)
+        Names of the required uncertainty input variables
+        ('exp_initial_input_var', 'impf_initial_input_var', 'haz_initial_input_var',
+         'exp_final_input_var', 'impf_final_input_var', 'haz_final_input_var'')
+    _metric_names : tuple(str)
+        Names of the impact output metrics
+        ('aai_agg', 'freq_curve', 'at_event', 'eai_exp')
+    """
+
+    _input_var_names = (
+        'exp_initial_input_var',
+        'impf_initial_input_var',
+        'haz_initial_input_var',
+        'exp_final_input_var',
+        'impf_final_input_var',
+        'haz_final_input_var',
+    )
+    """Names of the required uncertainty variables"""
+
+    _metric_names = (
+        'aai_agg',
+        'freq_curve',
+        'at_event',
+        'eai_exp'
+    )
+    """Names of the cost benefit output metrics"""
+
+    def __init__(
+        self,
+        exp_initial_input_var: Union[InputVar, Exposures],
+        impf_initial_input_var: Union[InputVar, ImpactFuncSet],
+        haz_initial_input_var: Union[InputVar, Hazard],
+        exp_final_input_var: Union[InputVar, Exposures],
+        impf_final_input_var: Union[InputVar, ImpactFuncSet],
+        haz_final_input_var: Union[InputVar, Hazard],
+    ):
+        """Initialize UncCalcImpact
+
+        Sets the uncertainty input variables, the impact metric_names, and the
+        units.
+
+        Parameters
+        ----------
+        exp_initial_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.Exposure
+            Exposure uncertainty variable or Exposure
```

Comment:
I think the docstrings year need a bit more clarity. It is not simply any Exposure. In the end, reported uncertainties will be final - initial impacts, which should be made clear here too. idem for haz and impfset.